### PR TITLE
 Bug 1840721 - We should prefer switching to open tabs vs opening a duplicate tab

### PR DESCRIPTION
**Pull request by t-p-white**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1840721

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/controller/RecentBookmarksController.kt [hunk 1]
```diff
@@ -46,15 +50,27 @@ class DefaultRecentBookmarksController(
     private val activity: HomeActivity,
     private val navController: NavController,
     private val appStore: AppStore,
+    private val selectTabUseCase: TabsUseCases.SelectTabUseCase,
+    private val tabs: List<TabSessionState> = activity.components.core.store.state.tabs,
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/controller/RecentBookmarksController.kt:54]: this is the only line that gives me a little pause. Could we instead just pass in `BrowserStore`, make it mandatory with no default value, and retrieve the tabs in `handleBookmarkClicked`?

**Review by MozillaNoah — APPROVED** [review]
LGTM. just left one comment

File: fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/controller/RecentBookmarksController.kt (hunk 1 above)
Comment (t-p-white) [reply fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/controller/RecentBookmarksController.kt:54]: @MozillaNoah yes, that's a good spot, thanks!

File: fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/controller/RecentBookmarksController.kt [hunk 2]
```diff
@@ -46,15 +49,27 @@ class DefaultRecentBookmarksController(
     private val activity: HomeActivity,
     private val navController: NavController,
     private val appStore: AppStore,
+    private val selectTabUseCase: TabsUseCases.SelectTabUseCase,
+    private val store: BrowserStore,
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/controller/RecentBookmarksController.kt:53]: nit
```suggestion
    private val appStore: AppStore,
    private val browserStore: BrowserStore,
    private val selectTabUseCase: TabsUseCases.SelectTabUseCase,
```

**Review by MozillaNoah — APPROVED** [review]
one minor nit, but thanks for making the change!

**Pull request merged**